main.rb

```diff
@@ -0,0 +1,118 @@
+module Enumerable
+  
+  # my each
+  def my_each
+    count = 0
+    while count < length
+      yield(self[count])
+      count += 1
+    end
+    self
+  end
+  
+  # my_each_with_each
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+    self
+  end
+  
+  # my_select
+  def my_select
+    count = 0
+    arr = []
+    while count < length
+      arr.push(self[count]) if yield(self[count])
+      count += 1
+    end
+    arr
+  end
+  
+  # my_all
+  def my_all?
+    count = 0
+    while count < length
+      count += 1
+      return false unless yield(self[count])
+      
+    end
+    true
+  end
+  
+  # my_any?
+  def my_any?
+    count = 0
+    while count < length
+      count += 1
+      return true if yield(self[count])
+    end
+    false
+  end
+  
+  # my_none?
+  def my_none?
+    count = 0
+    while count < length
+      false if yield(self[count])
+      count += 1
+    end
+    true
+  end
+  
+  # my_count
+  def my_count
```

Comment:
This method, whilst it returns the correct count while a block is being passed. It does not take an argument as the original `count` method. So if I run `array.my_count(1)` it would return an error because the method is not expecting arguments. While `array.count(1)` will count how many integers 1 are in the target array. Please do kindly fix this issue.